### ISSUE TITLE
SoundCloud crashes (403 error?)

### ISSUE THREAD

**Issue by brachna** [Miserlou/SoundScrape#166]
```
Microsoft Windows [Version 6.3.9600]
(c) Корпорация Майкрософт (Microsoft Corporation), 2013. Все права защищены.

C:\Users\MSI>soundscrape -f -p "D:/Music/1" "https://soundcloud.com/celldweller/electric-eye"
Active code page: 65001
Traceback (most recent call last):
  File "c:\python35\lib\site-packages\soundscrape\soundscrape.py", line 184, in process_soundcloud
    resolved = client.get('/resolve', url=artist_url, limit=200)
  File "c:\python35\lib\site-packages\soundcloud-0.5.0-py3.5.egg\soundcloud\client.py", line 133, in _request
  File "c:\python35\lib\site-packages\soundcloud-0.5.0-py3.5.egg\soundcloud\request.py", line 148, in make_request
  File "c:\python35\lib\site-packages\requests\models.py", line 909, in raise_for_status
    raise HTTPError(http_error_msg, response=self)
requests.exceptions.HTTPError: 403 Client Error: Forbidden for url: https://api.soundcloud.com/tracks/308914769?client_id=175c043157ffae2c6d5fed16c3d95a4c

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "c:\python35\lib\runpy.py", line 184, in _run_module_as_main
    "__main__", mod_spec)
  File "c:\python35\lib\runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "C:\Python35\Scripts\soundscrape.exe\__main__.py", line 9, in <module>
  File "c:\python35\lib\site-packages\soundscrape\soundscrape.py", line 119, in main
    process_soundcloud(vargs)
  File "c:\python35\lib\site-packages\soundscrape\soundscrape.py", line 198, in process_soundcloud
    hard_track_url = get_hard_track_url(item_id)
  File "c:\python35\lib\site-packages\soundscrape\soundscrape.py", line 518, in get_hard_track_url
    json_response = response.json()
  File "c:\python35\lib\site-packages\requests\models.py", line 866, in json
    return complexjson.loads(self.text, **kwargs)
  File "c:\python35\lib\site-packages\simplejson\__init__.py", line 516, in loads
    return _default_decoder.decode(s)
  File "c:\python35\lib\site-packages\simplejson\decoder.py", line 374, in decode
    obj, end = self.raw_decode(s)
  File "c:\python35\lib\site-packages\simplejson\decoder.py", line 404, in raw_decode
    return self.scan_once(s, idx=_w(s, idx).end())
simplejson.scanner.JSONDecodeError: Expecting value: line 1 column 1 (char 0)

C:\Users\MSI>
```

**Comment by Miserlou** [comment]
Confirmed. Probably need to get a new scrape client id.

**Comment by cowlicks** [comment]
Can I manually add a new scape client id?

I'm getting the same failure for https://soundcloud.com/postmalone/congratulations

**Comment by jchook** [comment]
Getting same error for https://soundcloud.com/muramasamusic/day-1

I tried two methods of download and got the same error each time:

```
soundscrape https://soundcloud.com/muramasamusic/day-1
soundscrape muramasamusic -t day-1
```

```
Traceback (most recent call last):
  File "/usr/local/lib/python3.5/dist-packages/soundscrape/soundscrape.py", line 163, in process_soundcloud
    resolved = client.get('/resolve', url=track_url, limit=200)
  File "/usr/local/lib/python3.5/dist-packages/soundcloud/client.py", line 133, in _request
    return wrapped_resource(make_request(method, url, kwargs))
  File "/usr/local/lib/python3.5/dist-packages/soundcloud/request.py", line 148, in make_request
    result.raise_for_status()
  File "/usr/lib/python3/dist-packages/requests/models.py", line 840, in raise_for_status
    raise HTTPError(http_error_msg, response=self)
requests.exceptions.HTTPError: 403 Client Error: Forbidden for url: https://api.soundcloud.com/tracks/168201561?client_id=175c043157ffae2c6d5fed16c3d95a4c

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/bin/soundscrape", line 11, in <module>
    load_entry_point('soundscrape==0.30.1', 'console_scripts', 'soundscrape')()
  File "/usr/local/lib/python3.5/dist-packages/soundscrape/soundscrape.py", line 119, in main
    process_soundcloud(vargs)
  File "/usr/local/lib/python3.5/dist-packages/soundscrape/soundscrape.py", line 198, in process_soundcloud
    hard_track_url = get_hard_track_url(item_id)
  File "/usr/local/lib/python3.5/dist-packages/soundscrape/soundscrape.py", line 519, in get_hard_track_url
    json_response = response.json()
  File "/usr/lib/python3/dist-packages/requests/models.py", line 808, in json
    return complexjson.loads(self.text, **kwargs)
  File "/usr/local/lib/python3.5/dist-packages/simplejson/__init__.py", line 518, in loads
    return _default_decoder.decode(s)
  File "/usr/local/lib/python3.5/dist-packages/simplejson/decoder.py", line 370, in decode
    obj, end = self.raw_decode(s)
  File "/usr/local/lib/python3.5/dist-packages/simplejson/decoder.py", line 400, in raw_decode
    return self.scan_once(s, idx=_w(s, idx).end())
simplejson.errors.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
```

**Comment by jchook** [comment]
Yes it appears to be an issue in the latest master.

```
~/dev/SoundScrape on master
➜ ./test.sh 
[################################] 2739/2739 - 00:00:01
[################################] 5856/5856 - 00:00:01
[################################] 1558/1558 - 00:00:00
[################################] 3253/3253 - 00:00:01
..
======================================================================
ERROR: test_soundcloud_hard (test.TestSoundscrape)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/jchook/dev/SoundScrape/tests/test.py", line 52, in test_soundcloud_hard
    process_soundcloud(vargs)
  File "/home/jchook/dev/SoundScrape/soundscrape/soundscrape.py", line 272, in process_soundcloud
    data = get_soundcloud_api2_data(artist_id)
  File "/home/jchook/dev/SoundScrape/soundscrape/soundscrape.py", line 499, in get_soundcloud_api2_data
    parsed = response.json()
  File "/home/jchook/.local/lib/python3.5/site-packages/requests/models.py", line 892, in json
    return complexjson.loads(self.text, **kwargs)
  File "/usr/local/lib/python3.5/dist-packages/simplejson/__init__.py", line 518, in loads
    return _default_decoder.decode(s)
  File "/usr/local/lib/python3.5/dist-packages/simplejson/decoder.py", line 370, in decode
    obj, end = self.raw_decode(s)
  File "/usr/local/lib/python3.5/dist-packages/simplejson/decoder.py", line 400, in raw_decode
    return self.scan_once(s, idx=_w(s, idx).end())
simplejson.errors.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
-------------------- >> begin captured logging << --------------------
urllib3.connectionpool: DEBUG: Starting new HTTPS connection (1): api.soundcloud.com
urllib3.connectionpool: DEBUG: https://api.soundcloud.com:443 "GET /resolve?limit=200&url=https%3A%2F%2Fsoundcloud.com%2Fpuptheband&client_id=a3dd183a357fcff9a6943c0d65664087 HTTP/1.1" 302 136
urllib3.connectionpool: DEBUG: https://api.soundcloud.com:443 "GET /users/41870548?client_id=a3dd183a357fcff9a6943c0d65664087 HTTP/1.1" 200 433
urllib3.connectionpool: DEBUG: Starting new HTTPS connection (1): api.soundcloud.com
urllib3.connectionpool: DEBUG: https://api.soundcloud.com:443 "GET /users/41870548/tracks?limit=200&client_id=a3dd183a357fcff9a6943c0d65664087 HTTP/1.1" 200 28
urllib3.connectionpool: DEBUG: Starting new HTTPS connection (1): api-v2.soundcloud.com
urllib3.connectionpool: DEBUG: https://api-v2.soundcloud.com:443 "GET /stream/users/41870548?limit=500&client_id=OmTFHKYSMLFqnu2HHucmclAptedxWXkq&app_version=1481046241 HTTP/1.1" 401 0
--------------------- >> end captured logging << ---------------------

----------------------------------------------------------------------
Ran 8 tests in 17.313s

FAILED (errors=1)
```

**Comment by SimplicityGuy** [comment]
Similar to #235, but also the other errors here are around poor HTTP error handling. Those are addressed in #241.
